### ISSUE TITLE
Stylelint - How to use if/else in SCSS

### ISSUE THREAD

**Issue by mattheu** [humanmade/coding-standards#160]
I want to use `@if` `@else if` and `@else` but I can't work out a way to use them without hitting linting errors.

The code below seems to be the expected formatting. However the rule `at-rule-empty-line-before` is flagging it as an error. If I try to add newlines before `@else if` and `@else` I get many more errors such as: `at-if-closing-brace-space-after`, `at-if-closing-brace-newline-after`, `at-else-empty-line-before` etc. 

```scss

@mixin text-style( $style: null ) {
	@if $style == 'heading' {
		font-family: $ing-font--heading;
		font-weight: $ing-font-weight;
		color: $ing-secondary;
	} @else if $style == 'subheading' {
		font-family: $ing-font;
		font-weight: $ing-font-weight--medium;
		color: $ing-secondary;
	} @else {
		font-family: $ing-font;
		font-weight: $ing-font-weight;
		color: $ing-secondary;
	}
}
```

If the above code should be allowed, the quickest fix is to add them as exceptions to the rule `at-rule-empty-line-before`

I currently have this in my local `.stylelintrc`

```json
{
	"extends": "@humanmade/stylelint-config",
	"plugins": "stylelint-scss",
	"rules": {
		"at-rule-empty-line-before": [
			"always", {
				"except": [ "blockless-after-blockless", "first-nested" ],
				"ignore": [ "after-comment" ],
				"ignoreAtRules": [ "if", "else if", "else" ]
			}
		],
	}
}

```

**Comment by mikeselander** [comment]
I would expect the above formatting to be what we'd prefer as well. This looks like a good addition to me.
